src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java

```diff
@@ -242,16 +242,19 @@ public static void addRecipe(IRecipe recipe)
         CraftingManager.getInstance().getRecipeList().add(recipe);
     }
 
-    public static void addSmelting(Block input, @Nonnull ItemStack output, float xp)
+    @Deprecated // Use FurnaceRecipeRegistry
```

Comment:
Also it looks like you removed a `@Nonnull` on the ItemStack parameter accidentally.